src/libraries/Microsoft.Extensions.HostFactoryResolver/src/HostFactoryResolver.cs

```diff
@@ -268,7 +284,7 @@ public object CreateHost()
                     // Wait before throwing an exception
                     if (!_hostTcs.Task.Wait(_waitTimeout))
                     {
-                        throw new InvalidOperationException("Unable to build IHost");
+                        throw new InvalidOperationException($"Timed out waiting for the entry point to build the IHost after {s_defaultWaitTimeout}.");
```

Comment:
I think we should mention the environment variable if we keep it.

```suggestion
                        throw new InvalidOperationException($"Timed out waiting for the entry point to build the IHost after {s_defaultWaitTimeout}. This timeout can be modified using the '{TimeoutEnvironmentKey}' environment variable.");
```